inspirehep/modules/pidstore/providers.py

```diff
@@ -84,8 +86,13 @@ def create(cls, object_type=None, object_uuid=None, **kwargs):
 
     @staticmethod
     def schema_to_pid_type(schema):
-        pid_type = os.path.splitext(os.path.basename(schema))[0]
-        if pid_type == 'hep':
+        endpoint = os.path.splitext(os.path.basename(schema))[0]
+        if endpoint == 'hep':
             # FIXME: temp hack until we rename hep.json to literature.json
-            return 'literature'
-        return pid_type
+            endpoint = 'literature'
+        return get_pid_type_for(endpoint)
+
+
+def get_pid_type_for(endpoint):
+    """Get the pid_type code corresponding to the recid of the given endpoint."""
+    return RECORDS_REST_ENDPOINTS[endpoint]['pid_type']
```

Comment:
In particular there is no good default, and it would hide wrong usages. I'd prefer to bubble up the exception to the calling code which is really aware of the origin of the `endpoint` argument and should know how to act on a `KeyError`.